var/spack/repos/builtin/packages/pmix/package.py

```diff
@@ -49,11 +52,34 @@ class Pmix(AutotoolsPackage):
 
     depends_on('libevent@2.0.20:2.0.22,2.1.8')
     depends_on('hwloc@1.11.0:1.11.99,2.0.1:', when='@3.0.0:')
+    depends_on("m4", type=("build"), when="@master")
+    depends_on("autoconf", type=("build"), when="@master")
+    depends_on("automake", type=("build"), when="@master")
+    depends_on("libtool", type=("build"), when="@master")
+    depends_on("perl", type=("build"), when="@master")
+
+    def autoreconf(self, spec, prefix):
+        """Only needed when building from git checkout"""
+        # If configure exists nothing needs to be done
+        if os.path.exists(self.configure_abs_path):
+            return
+        # Else bootstrap with autotools
+        perl = which('perl')
+        perl('./autogen.pl')
 
     variant('pmi_backwards_compatibility',
             default=True,
             description="Toggle pmi backwards compatibility")
 
+    variant('restful',
+            default=False,
+            description="allow a PMIx server to request services from "
+            "a system-level REST server")
```

Comment:
Is there a corresponding configure flag to enable/disable this feature?

Reply:
`--with-curl --with-libjansson` plus the usual `=<path>` for non-standard locations. Both options will automatically enable if the support is found in the standard locations, so I will likely add something like `--enable-restful-support` to make it easier to turn "off" if someone doesn't want it.